lib/rspec/openapi/schema_merger.rb

```diff
@@ -25,22 +25,23 @@ def normalize_keys(spec)
   # Also this needs to be aware of OpenAPI details unlike an ordinary deep_reverse_merge
   # because a Hash-like structure may be an array whose Hash elements have a key name.
   #
-  # TODO: Perform more intelligent merges like rerouting edits / merging types
   # TODO: Should we probably force-merge `summary` regardless of manual modifications?
   def deep_reverse_merge!(base, spec)
     spec.each do |key, value|
       if base[key].is_a?(Hash) && value.is_a?(Hash)
-        deep_reverse_merge!(base[key], value)
-      elsif !base.key?(key)
-        base[key] = value
+        if !base[key].key?("$ref")
+          deep_reverse_merge!(base[key], value)
+        end
       elsif base[key].is_a?(Array) && value.is_a?(Array)
         # parameters need to be merged as if `name` and `in` were the Hash keys.
         if key == 'parameters'
           base[key] |= value
           base[key].uniq! { |param| param.slice('name', 'in') }
+        else
+          base[key] = value
```

Comment:
I wondered if just overwriting the value of `base` is fine, but given that what we have as an array in the current test in this repository are just `servers`, `tags`, `parameters`, and `example`, it's probably okay to rewrite it, ignoring the possibility of the need to partially update arrays inside it. We may need to recursively do something if we find other array fields that are not compatible with this idea, but I don't come up with such a thing at this moment.